_drafts/conda.md

```diff
@@ -0,0 +1,698 @@
+before and after activating a Conda environment.
+
+```
+$ which make && make --version
+/usr/bin/make
+GNU Make 3.81
+
+# Activate our Conda environment
+$ conda activate example
+
+(example)
+$ which make && make --version
+/Users/tyler/miniconda3/envs/example/bin/make
+GNU Make 4.2.1
+```
+
+### Common Developer Environment Alternatives
+
+I have found Conda to be the least restrictive and easiest way for firmware
+teams manage developer environments, but let's go through the most popular
+alternatives.
+
+#### Docker
+
+Docker is the best way to create developer environments from scratch and for
+temporary purposes (such as in continuous integration). A `Dockerfile`, which
+commonly lives in a project's repo, defines any and all setup command, such as
+downloading and install system packages, installing Python requirements, and
+setting up permissions and volumes.
+
+Despite the numerous benefits, Docker is not a great fit for local developer
+environments.
+
+- By using Docker, you'll have two file systems, one on the host machine and one
+  within the container. Without a large investment in up-front tooling to help
+  developers manage the code, build artifacts, debugging tools, and volume
+  mounts, it is cumbersome and slow.
+- When using macOS or Windows and Docker, shared, mounted volumes have
+  file-sharing performance issues, which can cause build times to increase
+  between 2-10x depending on where the files live.[^2]
+- For developers that like the use of IDE's and other visual tools, a virtual
+  machine using VMWare or VirtualBox may be more appropriate.
+
+With all that said, using Docker in your CI system is a no-brainer. Conda and
+Docker can even be used _together_, where Docker manages the system image and
+low-level setup, and Conda manages all of the packages required to build and
+test the firmware.
+
+Check out [this previous Interrupt
+post]({% post_url 2019-09-17-continuous-integration-for-firmware %}) for how to
+get up and running quickly with Docker and continuous integration.
+
+#### Included System Package Managers
+
+System package managers, such as Brew, Apt, MinGW, etc., are very good at
+installing packages and managing dependencies between them. They are also great
+at ensuring that the modern software you use everyday works correctly, safely,
+and is kept up-to-date. They are not good at ensuring that the software or
+firmware that **you** build continues to work correctly. They are also not good
+at juggling multiple versions of packages that aren't the default ones.
+
+It is not uncommon for firmware consultancies or companies to have multiple
+different firmware projects in flight. This could be a legacy project and a
+newer one. I imagine the following wouldn't be far off:
+
+**Old Project Dependencies:** GNU Make 4.1, GCC 5.4.1, GDB 7.11, Python 3.5
+**New Project Dependencies:** GNU Make 4.2.1, GCC 8.3.1, GDB 8.3, Python 3.6
```

Comment:
You may need to add a `<br />` tag to force the line break. It's not rendered correctly on the blog at the moment.